h2/src/main/org/h2/table/TableLink.java

```diff
@@ -64,6 +64,7 @@
     private boolean globalTemporary;
     private boolean readOnly;
     private boolean targetsMySql;
+    private int fetchSize = -1;
```

Comment:
You you have a case to distinguish `setFetchSize(0)` and “`setFetchSize()` shouldn't be called”?

Because according to JDBC documentation, 0 should be default and it means no hint at all. If you don't know any drivers that violate the specification, there are no reasons to assign `-1` here and in other places `fetchSize != -1` needs to be replaced with `fetchSize != 0`.

Reply:
done